addons/io_scene_gltf2/blender/exp/gltf2_blender_extract.py

```diff
@@ -597,7 +597,10 @@ def extract_primitives(glTF, blender_mesh, blender_vertex_groups, modifiers, exp
 
             v = convert_swizzle_location(vertex.co, export_settings)
             if blender_polygon.use_smooth:
-                n = convert_swizzle_location(vertex.normal, export_settings)
+                if blender_mesh.has_custom_normals:
```

Comment:
@pjoe This is awesome, thanks!

I'm testing a strange case now, trying to reproduce this workflow: https://github.com/KhronosGroup/glTF-Blender-IO/pull/413#issuecomment-481539184

In this comment, a non-smoothed cube was beveled with "Harden Normals", producing some custom normals on a non-smooth mesh.  But it looks like your custom normal export fix here is hidden behind `if use_smooth` type check, is that intentional?  Should the custom normals take priority over the lack of smooth shading?

Reply:
Hmm, custom normals require the mesh's auto smooth to be active, and I think it also needs smooth shading to properly show up in the viewport. If you choose flat shading you are basically telling blender you want to use the face normals instead of vertex normals (or custom split normals)

Reply:
More hmmm, does seem to behave a bit weird though :S

Reply:
> If you choose flat shading you are basically telling blender you want to use the face normals instead of vertex normals (or custom split normals)

You can apparently use custom normals in flat shading (left). IIRC there are plans to improve the user interface for editing custom normals. This may be useful for some effects.

![image](https://user-images.githubusercontent.com/4047289/55940452-77423200-5c40-11e9-9885-75ff172991f4.png)
![image](https://user-images.githubusercontent.com/4047289/55940510-92ad3d00-5c40-11e9-8eb1-5298188f4926.png)

Reply:
Aye, should move the custom normal code out of the `use_smooth`check then